File: Week3 Daily 2/app/src/main/java/com/ralu/week3daily2/MainActivity.java
package com.ralu.week3daily2;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListAdapter;
import android.widget.ListView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = "Main";
    private ListView lvMain;
    private List<String> celebNames = new ArrayList<>();




    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // binding data

        lvMain = findViewById(R.id.lvMain);

        // populate the list

        celebNames.add("Michelle Pfeiffer");
        celebNames.add("Tom Cruise");
        celebNames.add("Lionel Messi");
        celebNames.add("Freddy Mercury");
        celebNames.add("Queen Elisabeth");


        // link the String list to a layout in an adapter
        ListAdapter listAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, celebNames);
        // link the adapter to the listview
        lvMain.setAdapter(listAdapter);

        // set onclick listner

        lvMain.setOnItemClickListener(
                new AdapterView.OnItemClickListener() {
                    @Override
                    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                        Intent intent = new Intent (MainActivity.this, Celebrity_detail.class);
                        intent.putExtra("position", position);
                        startActivity(intent);
                    }
                }

        );






    }




}

File: SpaceStation/app/src/main/java/com/example/raluc/spacestation/ui/MainActivity.kt
package com.example.raluc.spacestation.ui

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.content.BroadcastReceiver
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import com.example.raluc.spacestation.R
import com.google.android.gms.location.FusedLocationProviderClient
import com.google.android.gms.location.LocationServices

class MainActivity : AppCompatActivity() {

    private val TAG = "Main"

    lateinit var mainViewModel: MainViewModel
    lateinit var recycler: RecyclerView
    lateinit var layoutManager: RecyclerView.LayoutManager
    lateinit var client : FusedLocationProviderClient
    lateinit var broadcastReceiver: BroadcastReceiver

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.d(TAG, "onCreate")

        mainViewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
        recycler = findViewById(R.id.recyclerView)
        client = LocationServices.getFusedLocationProviderClient(this)


        mainViewModel.getStation().observe(this, Observer {
            recycler.layoutManager = LinearLayoutManager(this)
            recycler.adapter = it
            Log.d(TAG, "main created")
        })


    }
}
